src/status_im2/contexts/chat/lightbox/zoomable_image/view.cljs

```diff
@@ -204,76 +203,61 @@
-                                            rescale
-                                            set-full-height?))
-       (utils/handle-zoom-out-signal zoom-out-signal index (anim/get-val (:scale animations)) rescale)
-       [:f>
-        (fn []
-          (let [tap (tap-gesture #(on-tap portrait?))
-                double-tap
-                (double-tap-gesture dimensions animations rescale transparent? #(on-tap portrait?))
-                pinch
-                (pinch-gesture dimensions animations props rescale transparent? #(on-tap portrait?))
-                pan-x (pan-x-gesture dimensions animations props rescale)
-                pan-y (pan-y-gesture dimensions animations props rescale)
-                composed-gestures (gesture/exclusive
-                                   (gesture/simultaneous pinch pan-x pan-y)
-                                   (gesture/exclusive double-tap tap))]
-            [gesture/gesture-detector {:gesture composed-gestures}
-             [reanimated/view
-              {:style (style/container dimensions
-                                       animations
-                                       @set-full-height?
-                                       (= curr-orientation orientation/portrait))}
-              [reanimated/fast-image
-               {:source    {:uri (:image content)}
-                :native-ID (when focused? :shared-element)
-                :style     (style/image dimensions animations (:border-value args))}]]]))]))])
+  [{:keys [image-width image-height content message-id]} index render-data]
+  (let [state                      (utils/init-state)
+        shared-element-id          (rf/sub [:shared-element-id])
+        exit-lightbox-signal       (rf/sub [:lightbox/exit-signal])
+        zoom-out-signal            (rf/sub [:lightbox/zoom-out-signal])
+        curr-orientation           (or (rf/sub [:lightbox/orientation])
+                                       orientation/portrait)
+        {:keys [set-full-height?]} render-data
+        focused?                   (= shared-element-id message-id)
+        dimensions                 (utils/get-dimensions
+                                    (or image-width c/default-dimension)
+                                    (or image-height c/default-duration)
+                                    curr-orientation
+                                    render-data)
+        animations                 (utils/init-animations)
+        rescale                    (fn [value exit?]
+                                     (utils/rescale-image value
+                                                          exit?
+                                                          dimensions
+                                                          animations
+                                                          state))]
+    (rn/use-effect (fn []
+                     (js/setTimeout #(reset! set-full-height? true) 500)))
```

Comment:
We should cancel the timer on unmount. I see two reasons to basically always try to clean up timers:

- It's easy for some developer in the future to make changes to the timer's delays, and then suddenly :boom: 2+ scheduled timers will race against each other and cause weird UI updates. Although rare, this can happen.
- Cleaning up all timers help save CPU cycles, because if an unmount happens before certain timers are triggered, they'll be deleted and no needless state changes will happen.